langchain/vectorstores/chroma.py

```diff
@@ -352,26 +352,28 @@ def persist(self) -> None:
             )
         self._client.persist()
 
-    def update_document(self, document_id: str, document: Document) -> None:
+    def update_document(
```

Comment:
this isnt backwards compatible at all - can we either allow for `Union[str, List[str]]` or have a separate method?

Reply:
Both document_id and document can be `Union[str, List[str]]` but we need to do type check. Because embed_documents and update functions takes List for text and document_ids variables. I added a typecheck like that:
```

def update_document(
        self, document_ids: Union[str, List[str]], documents: Union[Document, List[Document]]
    ) -> None:
        """Update a document in the collection.

        Args:
            document_ids (Union[str, List[str]]): IDs of the document to update.
            documents (Union[Document, List[Document]]): Documents to update.
        """
        if isinstance(document_ids, str):
            document_ids = [document_ids]  # Convert to a list if it's a string
        
        if isinstance(documents, Document):
            documents = [documents]  # Convert to a list if it's a Document

        text = [document.page_content for document in documents]
        metadata = [document.metadata for document in documents]
        if self._embedding_function is None:
            raise ValueError(
                "For update, you must specify an embedding function on creation."
            )
        embeddings = self._embedding_function.embed_documents(text)

        self._collection.update(
            ids=document_ids,
            embeddings=embeddings,
            documents=text,
            metadatas=metadata,
        )
```
If you say it's not suitable, a separate method can also be written. But I think #5415 already does that as @Satyam-79 mentioned. However, if you still want me to write it, I can write it and push it to this pull request.